File: swift/problem415.swift
class Solution {
    func addStrings(_ num1: String, _ num2: String) -> String {
        if (num1.count == 0 || num2.count == 0) {
            return (num1+num2)
        }
        
        var num1 = num1.map{Int(String($0))!}
        var num2 = num2.map{Int(String($0))!}
        if (num1.count < num2.count) {
            swap(&num1,&num2)
        }
        
        let longer = num1.count; let shorter = num2.count
        var carry = 0; var ans = String()
        
        for i in 1...shorter {
            let sum = num1[longer-i]+num2[shorter-i]+carry
            ans = String(sum%10) + ans
            carry = sum/10
        }
        
        for i in (0..<(longer-shorter)).reversed() {
            let sum = num1[i] + carry
            ans = String(sum%10) + ans
            carry = sum/10
        }
        
        if (carry != 0) {
            ans = String(carry) + ans
        }
        
        return ans
    }
}

File: python/problem415.py
class Solution:
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
          
        if len(num1)<len(num2):
            num1,num2=num2,num1
        num2='0'*(len(num1)-len(num2))+num2
        
        r=''
        m='0'
        l=len(num1)
        for i in range(l):
            k=l-1-i
            
            j = ord(num1[k])-ord('0') + ord(num2[k])-ord('0') + ord(m)-ord('0')
            
            m='0'
            if j>=10:
                j-=10
                m='1'
            r+=chr(j+ord('0'))
        if m=='1':
            r+=m   
        return r[::-1]
        